client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/failover/strategies/FailoverStrategy.java

```diff
@@ -0,0 +1,59 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.kaaproject.kaa.client.channel.failover.strategies;
+
+import org.kaaproject.kaa.client.channel.failover.FailoverDecision;
+import org.kaaproject.kaa.client.channel.failover.FailoverStatus;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Failover strategy is responsible for producing failover decisions based on failover statuses.
+ */
+public interface FailoverStrategy {
```

Comment:
FailoverManager is aware of recovery so it's possible to let FailoverManager notify strategy about that. We could also pass connection information to strategy.
Let me know if it's what you've been looking for.